src/plugins/data/server/search/search_service.ts

```diff
@@ -162,14 +162,6 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
             .search(requestData, options)
             .pipe(
               first(),
-              map((response) => {
```

Comment:
Why did you choose to move the shimming back to the individual strategies?

Reply:
I've been thinking about this more and I think it makes more sense in the strategy itself. Here's why:

Putting it on the route assumes that *all* search strategies will need to have the hits total shimmed. Since a custom search strategy might not even have a "hits" portion, I don't think it makes sense to have this in the route.

Also, if we do include it at the route and not in the strategy itself, then things like `SearchSource` will have to manually shim it (like they were doing prior to this PR)... Otherwise you'll get a different response if you're using search source on the client vs. server. This is also a concern I have with it in the route regarding the strategy itself: If you're using the client APIs, you'll get a different response than the server APIs, which I think we should generally try to avoid. Thoughts?